src/pip/_internal/cli/req_command.py

```diff
@@ -48,24 +50,28 @@
 
 
 def _create_truststore_ssl_context() -> Optional["SSLContext"]:
+    # Truststore only works with Python 3.10+
     if sys.version_info < (3, 10):
-        raise CommandError("The truststore feature is only available for Python 3.10+")
+        return None
 
     try:
         import ssl
     except ImportError:
         logger.warning("Disabling truststore since ssl support is missing")
         return None
 
+    # Since truststore is developed with only Python 3.10+ in mind
+    # we delay the import until we know we're running pip with Python 3.10+.
     try:
-        import truststore
-    except ImportError:
-        raise CommandError(
-            "To use the truststore feature, 'truststore' must be installed into "
-            "pip's current environment."
-        )
+        from pip._vendor import truststore
+    # Truststore doesn't work on macOS versions earlier than 10.8
+    except OSError:
+        logger.warning("Disabling truststore since OS version isn't supported")
+        return None
```

Comment:
Is it possible to do better detection? OSError seems too coarse.

Reply:
Right now `truststore` raises an `OSError` when the macOS APIs that we need aren't available (ie macOS 10.8). Maybe we can change that error to `ImportError` instead? cc @davisagli

Reply:
@sethmlarson Or maybe a custom subclass of ImportError? It'd be nice to be able to distinguish between an expected condition (not supported) and an ImportError due to something being messed up with the installation of the vendored package.

Reply:
So @davisagli and I discussed this and unfortunately because this has to happen at import time we couldn't go the route of a custom exception, so we decided to standardize on raising an `ImportError` whenever the module can't support the current platform/Python version (currently only macOS 10.7 and earlier raise this error).

Hope this works for you, let me know your thoughts.